### PR TITLE
Adds `impl` formatting to `sfv2`

### LINKED ISSUE #2068: Add `impl` formatting to `sway-fmt-v2`

### PR DESCRIPTION
**Pull request by eureka-cpu**
Closes #2068 

- [x] Add better test

### CODE REVIEW

File: sway-fmt-v2/src/items/item_impl.rs [hunk 1]
```diff
@@ -1,20 +1,101 @@
...
+        line: &mut FormattedCode,
+        formatter: &mut Formatter,
+    ) -> Result<(), FormatterError> {
+        let brace_style = formatter.config.items.item_brace_style;
+        let extra_width = formatter.config.whitespace.tab_spaces;
+        let mut shape = formatter.shape;
+        let open_brace = Delimiter::Brace.as_open_char();
+        match brace_style {
+            ItemBraceStyle::AlwaysNextLine => {
+                // Add openning brace to the next line.
```
Comment (sezna) [inline sway-fmt-v2/src/items/item_impl.rs:64]:
```suggestion
                // Add opening brace to the next line.
```

File: sway-fmt-v2/src/fmt.rs [hunk 2]
```diff
@@ -680,4 +680,48 @@ trait Programmer {
             Formatter::format(&mut formatter, Arc::from(sway_code_to_format), None).unwrap();
         assert_eq!(correct_sway_code, formatted_sway_code)
     }
+    #[test]
+    fn test_impl() {
+        let sway_code_to_format = r#"script;
```
Comment (sezna) [inline sway-fmt-v2/src/fmt.rs:685]: The code to format and the corrected sway code are the same, is that intentional?

**Review by sezna — CHANGES_REQUESTED** [review]
RC until testing question is resolved

File: sway-fmt-v2/src/fmt.rs (hunk 2 above)
Comment (eureka-cpu) [reply sway-fmt-v2/src/fmt.rs:685]: Ah, thanks I forgot to make it messy!

**Review by sezna — APPROVED** [review]
I think we could add some more tests here, but this is good for now. Maybe in the future we can worry about a larger formatter test suite like the sway compiler one.

**Comment by kayagokalp** [comment]
this looks good to me but there seem to be some conflicts. I will happily approve once those are resolved 👍

**Review by sezna — DISMISSED** [review]

**Review by sezna — APPROVED** [review]

**Review by kayagokalp — APPROVED** [review]

**Pull request merged**